crates/seal-proxy/src/client.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c), Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::metrics_push::{metrics, push_metrics_to_prometheus};
+use axum::{extract::Extension, routing::get, Router};
+use prometheus::Registry;
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use serde_with::DurationSeconds;
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::time::Duration;
+use tokio::net::TcpListener;
+use tokio::task::JoinHandle;
+use tokio_util::sync::CancellationToken;
+
+pub const METRICS_ROUTE: &str = "/metrics";
+
+#[serde_as]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct MetricsPushConfig {
+    /// The interval of time we will allow to elapse before pushing metrics.
+    #[serde_as(as = "DurationSeconds<u64>")]
+    #[serde(
+        rename = "push_interval_secs",
+        default = "push_interval",
+        skip_serializing_if = "is_push_interval_default"
+    )]
+    pub push_interval: Duration,
+    /// The URL that we will push metrics to.
+    pub push_url: String,
+    /// Static labels to provide to the push process.
+    #[serde(default, skip_serializing_if = "is_none")]
+    pub labels: Option<HashMap<String, String>>,
+}
+
+#[derive(Clone, Debug)]
+pub struct EnableMetricsPush {
+    pub config: MetricsPushConfig,
+    pub cancel: Option<CancellationToken>,
+    pub bearer_token: String,
+}
+
+/// Configure the default push interval for metrics.
+pub fn push_interval() -> Duration {
+    Duration::from_secs(60)
+}
+
+/// Returns true if the `duration` is equal to the default push interval for metrics.
+pub fn is_push_interval_default(duration: &Duration) -> bool {
+    duration == &push_interval()
+}
+
+/// Returns true iff the value is `None` and we don't run in test mode.
+pub fn is_none<T>(t: &Option<T>) -> bool {
+    // The `cfg!(test)` check is there to allow serializing the full configuration, specifically
+    // to generate the example configuration files.
+    !cfg!(test) && t.is_none()
+}
+
+// Creates a new http server that has as a sole purpose to expose
+// and endpoint that prometheus agent can use to poll for the metrics.
+// A RegistryService is returned that can be used to get access in prometheus Registries.
+pub fn start_prometheus_server(addr: SocketAddr) -> Registry {
```

Comment:
is this one needed? today we get that from mysten_service::metrics::start_prometheus_server